modules/runner/src/integrationTest/java/org/apache/ignite/internal/compute/ItLogicalTopologyTest.java

```diff
@@ -115,21 +131,38 @@ void receivesLogicalTopologyEventsCausedByNodeRestart() throws Exception {
 
         restartNode(1);
 
-        waitForCondition(() -> events.size() >= 2, 10_000);
+        // There's a race between a new version of the node being validated, while the old version has not yet been removed from the logical
```

Comment:
Why is there such a race? Can't we always remove previous node version before making its new version as validated?